starters/http/dotnet/infra/core/monitor/appinsights-access.bicep

```diff
@@ -0,0 +1,21 @@
+param principalID string
+param roleDefinitionID string
+param appInsightsName string
+
+resource applicationInsights 'Microsoft.Insights/components@2020-02-02' existing = {
+  name: appInsightsName
+}
+
+// Allow access from API to app insights using a managed identity and least priv role
+resource appInsightsRoleAssignment 'Microsoft.Authorization/roleAssignments@2020-04-01-preview' = {
+  name: guid(applicationInsights.id, principalID, roleDefinitionID)
+  scope: applicationInsights
```

Comment:
nicely done scoping just to the resource!

Reply:
:-)